care/facility/api/viewsets/bed.py

```diff
@@ -57,6 +59,23 @@ class BedViewSet(
     search_fields = ["name"]
     filterset_class = BedFilter
 
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        number_of_beds = serializer.validated_data.get('number_of_beds')
+
+        # Bulk creating n number of beds
+        if number_of_beds > 1:
+            data = [serializer.validated_data] * number_of_beds
+            for i, d in enumerate(data):
+                d['name'] = d['name'] + f' - {i}'
+            res = Bed.objects.bulk_create(data)
+            return Response(res, status=status.HTTP_201_CREATED)
+        
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```

Comment:
you can do this in the serializer itself, override the create method

Reply:
@sainak 
The reason I chose to implement this here, was that serializer's create method also performs a number of checks. Those checks could help in debugging if the need be.